src/index.js

```diff
@@ -15,26 +17,77 @@ const traverse = require('traverse');
 const DEFAULT_REPLACEMENT = '--REDACTED--';
 
 /**
- * Module exports.
+ * Validate serializers.
  */
 
-module.exports = (
+function validateSerializers(serializers) {
+  serializers.map(({ path, serializer }) => {
+    if (typeof serializer !== 'function') {
```

Comment:
Well, this would be useful only to inform the user since the problem would still remain. An async function if invoked without await returns a Promise. The user would still be able to pass a serializer than returns a Promise and is not an async function.